specs/testing.md

```diff
@@ -1,632 +1,241 @@
+- Tests the actual HTTP boundary that production uses
+- Slow but comprehensive (< 30s per test)
 
-  it('should match files in directory', async () => {
-    // Pattern: projects/**
-  });
+### Runtime Strategy
 
-  it('should match files with name pattern', async () => {
-    // Pattern: **/*meeting*
-  });
+**Current: Node.js + Jest**
 
-  it('should handle empty results', async () => {});
+All tests run in Node.js because our dependencies (MCP SDK, AWS SDK) use CommonJS modules with JSON imports that workerd doesn't support. This works because:
+- Unit/integration tests don't need Worker-specific APIs
+- E2E tests run actual Worker via `wrangler dev`
+- Production bundler (esbuild) resolves CommonJS→ESM at build time
 
-  it('should enforce max_results limit', async () => {});
+**Future: Workers Runtime**
 
-  it('should reject invalid patterns', async () => {
-    // Test malformed globs
-  });
+If dependencies migrate to pure ESM or we pre-bundle for testing, we could use `@cloudflare/vitest-pool-workers` to test in actual workerd. This would catch platform-specific issues earlier, but the current gap is acceptable—E2E tests provide sufficient runtime validation.
 
-  it('should return metadata', async () => {
-    // Test size, modified date
-  });
-});
-```
+### Mock Strategy
 
-### Tool: `grep`
+Mocks should mirror API contracts without implementing internal logic. Keep them simple, predictable, and focused on enabling test scenarios (success paths, error injection, edge cases). Never mock third-party library internals—mock at the boundaries where we call them.
 
-```typescript
-describe('grep tool', () => {
-  it('should find matches across all files', async () => {});
+**Infrastructure mocks** (R2, KV): In-memory Map-based implementations matching Cloudflare API contracts.
 
-  it('should scope search to path', async () => {
-    // Test path: projects/**
-  });
+**Service mocks** (OAuth, GitHub): Controlled flows with configurable responses for testing authorization, error handling, and edge cases.
 
-  it('should return context lines', async () => {
-    // Test context_lines: 2
-  });
-
-  it('should handle regex patterns', async () => {
-    // Test: \\buser\\s+research\\b
-  });
-
-  it('should be case-insensitive by default', async () => {});
-
-  it('should enforce max_matches limit', async () => {});
-
-  it('should reject invalid regex', async () => {});
+---
 
-  it('should handle multiline matches', async () => {});
-});
-```
+## Coverage Standards
 
-### OAuth Handler
+### Trend-Based Monitoring
 
-```typescript
-describe('OAuth handler', () => {
-  it('should complete OAuth flow', async () => {});
+Use automated coverage tracking (Codecov, Coveralls, or similar) to monitor trends across commits. Visualize coverage history to identify patterns. Fail CI on significant drops unless manually overridden with justification. This creates accountability without mandating arbitrary absolutes.
```

Comment:
Can we avoid a service and just use something like this instead - https://github.com/clearlyip/code-coverage-report-action?